tests/test_ai/test_priority_algorithm/test_mutations.py

```diff
@@ -0,0 +1,263 @@
+import unittest
+from typing import List, Dict
+
+from ai.priority_algorithm.interfaces import Priority
+from ai.priority_algorithm.mutations import mutate_robinhood
+from ai.priority_algorithm.priority_teamset import PriorityTeamSet, PriorityTeam
+from models.student import Student
+from models.team import Team
+
+
+class TestPriority(Priority):
+    """
+    A custom priority for this test.
+    It wants students 1 and 2 to be in the same team
+    """
+
+    def validate(self):
+        return True
+
+    def satisfied_by(self, students: List[Student]) -> bool:
+        ids = [student.id for student in students]
+        return 1 in ids and 2 in ids
+
+
+def print_priority_team_set(priority_team_set: PriorityTeamSet):
+    """
+    For debugging purposes
+    """
+    for priority_team in priority_team_set.priority_teams:
+        print(f"Team {priority_team.team.id}: {priority_team.student_ids}", end=" ")
+    print()
+
+
+def equal_priority_team_sets(a: PriorityTeamSet, b: PriorityTeamSet) -> bool:
+    """
+    Checks if two priority team sets are equal
+    """
+    if len(a.priority_teams) != len(b.priority_teams):
+        return False
+    for priority_team_a, priority_team_b in zip(a.priority_teams, b.priority_teams):
+        if priority_team_a.team.id != priority_team_b.team.id:
+            return False
+        if priority_team_a.student_ids != priority_team_b.student_ids:
+            return False
+    return True
+
+
+def create_new_priority_team_set(
+    num_teams: int, num_students: int
+) -> (PriorityTeamSet, Dict[int, Student]):
```

Comment:
nit;
You can import `Tuple` from the `typing` library instead of just using parentheses (it's supposed to be better but I forgot why, either way, it's more explicit)